internal/api/external.go

```diff
@@ -491,9 +491,20 @@ func (a *API) loadExternalState(ctx context.Context, r *http.Request) (context.C
 	}
 	config := a.config
 	claims := ExternalProviderClaims{}
-	p := jwt.NewParser(jwt.WithValidMethods([]string{jwt.SigningMethodHS256.Name}))
+	p := jwt.NewParser(jwt.WithValidMethods(config.JWT.ValidMethods))
 	_, err := p.ParseWithClaims(state, &claims, func(token *jwt.Token) (interface{}, error) {
-		return []byte(config.JWT.Secret), nil
+		if kid, ok := token.Header["kid"]; ok {
+			if kidStr, ok := kid.(string); ok {
+				return conf.FindPublicKeyByKid(kidStr, &config.JWT)
+			}
```

Comment:
Feels like this whole section is repeating a bit? Maybe it can be extracted somewhere?

Reply:
hmm would prefer to leave this here for now because the only other place we do it is in here: https://github.com/supabase/auth/blob/dc2391d15f2c0725710aa388cd32a18797e6769c/internal/api/auth.go#L81-L84